packages/nimble-components/src/patterns/link/styles.ts

```diff
@@ -0,0 +1,69 @@
+import { css } from '@microsoft/fast-element';
+import { display } from '@microsoft/fast-foundation';
+import {
+    linkActiveDisabledFontColor,
+    linkActiveFont,
+    linkActiveFontColor,
+    linkActiveProminentFont,
+    linkActiveProminentFontColor,
+    linkDisabledFontColor,
+    linkFont,
+    linkFontColor,
+    linkProminentFont,
+    linkProminentFontColor,
+    linkProminentDisabledFontColor
+} from '../../theme-provider/design-tokens';
+
+export const linkStyles = css`
+    ${display('inline')}
+
+    :host {
```

Comment:
As I go through this file it looks like tons of these properties are being overridden by the breadcrumb-item.

Can you trim this down to what is actually useful for sharing? I'll let you do the evaluation but it may be that the breadcrumb-item's pattern for being controlled by it's parent breadcrumb means that it's not worth sharing styles. If you find that to be the case that's fine. Right now I can't see the leverage being gained by sharing if most of the properties are just overridden immediately.

Reply:
I'm not sure I'm seeing what you're seeing. I got rid of two properties we were setting on the :host in breadcrumb-item (box-sizing and font) that weren't needed. The only things I see left that are being overwritten is `display` on the `.control` and `color` on `control:active`. That doesn't seem excessive/problematic to me. Am I misunderstanding?

Reply:
You can see the styles being overriden more easily if you look at storybook:

![image](https://user-images.githubusercontent.com/1588923/207103322-3d02031b-8332-4993-aced-1e9742fae759.png)

You can see that for a breadcrumb item the shared styles for the `.control` selector are all overridden by other styles in breadcrumb. Non of them actually impact the rendering of the control (they are all crossed out) so there is no point in sharing that style.

Reply:
Just to point out another example, breadcrumb-item does not have an appearance attribute. So all the styles targeting the appearance attribute selector don't make sense in a shared stylesheet, They do not apply to the breadcrumb item.

Reply:
I moved the appearance-based styles and the `.control`'s `display` attribute out of the shared style. I think that's all that should be done. The other two overridden styles are conditional -- because of the `:any-link` -- so they should not be moved.

Reply:
I disagree and I think there is a conceptual issue that will help clarify. The goal of our styling is to [rely on the cascade](https://github.com/ni/nimble/blob/main/packages/nimble-components/docs/css-guidelines.md#prefer-cascade-for-state-changes) for state changes. Right now the styles are written as if there are three states for the link disabled state `.control`, `.control:any-link` and `.control:not(:any-link)`.

But that doesn't actually make sense. The disabled state is binary so the styles are written incorrectly. Instead there should just be `.control {}` which represents the default state of the control which is link enabled and then `.control:not(:any-link)` which represents the style changes for the disabled state of the control.

So we should only see variations on:
```
.control {

}

.control:not(:any-link) {
}
```

We should not see any `.control:any-link` in the styles.

Reply:
I ended up removing the shared code altogether in this branch: https://github.com/ni/nimble/compare/users/makinc/anchor...users/makinchttps://ni.visualstudio.com/_pulls/anchor-mraj-proposal?expand=1

Some reasons:
- color and font are handled uniquely by the breadcrumb, they are defined by the parent
- the disabled states don't apply to the breadcrumb, not in the template that the `.control` element is removed altogether when href is "disabled"
- the cursor states don't really need to be manually defined. Disabling the href makes the browser handle the cursor state already
- That just leaves the text-decoration state being shared which seems like unnecessary shared code so I copied that per control

Reply:
Thanks for helping me out with this. I've adopted your changes and made sure everything still works as expected in Storybook.